### PR TITLE
implement 'budgetAccount' entities displaying

### LINKED ISSUE #30: Implement `BudgetAccount` entities displaying
In order to visualize current state of Budget for the user - we need to implement UI / UX to view budget accounts on "Budget" page

Implement fetching user's budget accounts during SSR and displaying the list, divided in 4 sections (rows with cards)

- Row with Income accounts
- Row with Debits, Credits, and Debts
- Row with Spending Categories 

These rows should display each Budget Account in a card with title, limit and icon. Each card should be 150 px width and 150 px height. Other than that - just use `Card` component with proper underlying structure (`CardHeader`, `CardContent`)

Depends on #23

### PR DESCRIPTION
**Pull request by Artemh59**
Closes #30

### CODE REVIEW

File: src/app/api/budgetAccount/get/route.ts
Comment (mudrila) [inline src/app/api/budgetAccount/get/route.ts:1]: We don't need this API route - we can fetch directly from DB during server-side rendering

File: src/app/components/molecules/BudgetAccount/BudgetAccountForm.tsx [hunk 1]
```diff
@@ -64,15 +77,105 @@ export default function BudgetAccountForm() {
     });
   };
 
+  useEffect(() => {
```
Comment (mudrila) [inline src/app/components/molecules/BudgetAccount/BudgetAccountForm.tsx:80]: Then again - we don't need this `useEffect`

File: src/app/components/molecules/BudgetAccount/BudgetAccountForm.tsx [hunk 2]
```diff
@@ -64,15 +77,105 @@ export default function BudgetAccountForm() {
...
+        {[
+          { accounts: budgetAccounts?.incomeAccounts, title: 'Income Accounts' },
+          {
+            accounts: budgetAccounts?.debtCreditDebitGoalAccounts,
+            title: 'Debt/Credit/Debit/Goal Accounts',
+          },
+          { accounts: budgetAccounts?.spendingCategories, title: 'Spending Categories' },
+        ].map((group, index) => {
+          return (
+            <Box key={index}>
```
Comment (mudrila) [inline src/app/components/molecules/BudgetAccount/BudgetAccountForm.tsx:133]: Please extract it into `BudgetAccountList` component and render it alongside with `BudgetAccountForm`, not within it.
Best path forward will be creating `src/components/pages/Budget/Budget.tsx` and then within that component

```
export default function Budget({ accounts }: { accounts: BudgetAccount[] }) {
  return (
    <Box>
      <Box sx={{ display: 'flex' ... }}>
        <Typography>Blah blah page title</Typography>
        <BudgetAccountForm />
      </Box>
      <Box sx={{ mt: 2 }}>
        <BudgetAccountSList accounts={accounts} />
      </Box>
  </Box>
  );
}
```

Obviously this is not complete implementation - but kinda an idea. Then you'll need to perform server-side rendering from the `src/app/budget/page.tsx`, fetch accounts and pass that in into newly create `Budget` component

File: src/app/components/molecules/IconSelect/IconSelect.tsx [hunk 3]
```diff
@@ -23,7 +23,12 @@ interface IconSelectProps {
   setIcon: (e: string) => void;
 }
 
-export default function IconSelect({ icon, setIcon }: IconSelectProps) {
+export function IconRenderrer({ iconName }: { iconName: string }) {
```
Comment (mudrila) [inline src/app/components/molecules/IconSelect/IconSelect.tsx:26]: Oh, good catch - haven't noticed it was within `IconSelect` previously.
Defining components shouldn't ever happen within another component - only outside as a separate high-level function

**Review by mudrila — CHANGES_REQUESTED** [review]
Please, remove API route and client-side fetching of `budgetAccounts` - we should be fetching that during SSR

https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#fetching-data-on-the-server-with-an-orm-or-database - here's an example on how to do that - ofc you need to apply our context (`prisma`, `session`, etc)

File: src/app/app/budget/page.tsx [hunk 4]
```diff
@@ -1,10 +1,32 @@
...
 
-export default function BudgetPage() {
-  return <BudgetAccountForm />;
+const convertDecimalToNumber = (account: BudgetAccount): SerializedBudgetAccount => ({
+  ...account,
+  monthlyLimit: Number(account.monthlyLimit),
+  currentBalance: Number(account.currentBalance),
+});
+
+async function getAccounts(): Promise<SerializedBudgetAccount[]> {
```
Comment (mudrila) [inline src/app/app/budget/page.tsx:18]: No need for this function extraction - SSR pages are good "containers" for such logic

File: src/app/app/budget/page.tsx [hunk 5]
```diff
@@ -1,10 +1,32 @@
...
+import { auth } from '@/utils/auth';
+import prisma from '@/utils/prisma';
 
 export const metadata: Metadata = {
   title: 'Budget | Fin-Plan',
 };
 
-export default function BudgetPage() {
-  return <BudgetAccountForm />;
+const convertDecimalToNumber = (account: BudgetAccount): SerializedBudgetAccount => ({
```
Comment (mudrila) [inline src/app/app/budget/page.tsx:12]: Please extract this into `src/utils/formatters.ts` and rename to `serializeBudgetAccount`

File: src/app/components/molecules/BudgetAccount/BudgetAccountForm.tsx [hunk 6]
```diff
@@ -14,7 +14,7 @@ import {
 } from '@mui/material';
 import { useState, useTransition } from 'react';
 import { toast } from 'sonner';
-import IconSelect from '@/app/components/molecules/IconSelect/IconSelect';
+import { IconSelect } from '@/app/components/molecules/IconSelect/IconSelect';
```
Comment (mudrila) [inline src/app/components/molecules/BudgetAccount/BudgetAccountForm.tsx:17]: Please convert `IconSelect` to be default export from `IconSelect.tsx` and revert this change

File: src/app/components/molecules/BudgetAccount/BudgetAccountList.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
+'use client';
+import { Box, Card, CardContent, CardHeader, Divider, Typography } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { IconRenderrer } from '@/app/components/molecules/IconSelect/IconSelect';
+import { SerializedBudgetAccount } from '@/constants/content';
+
+export default function BudgetAccountList({ accounts }: { accounts: SerializedBudgetAccount[] }) {
+  const [budgetAccounts, setBudgetAccounts] = useState<{
+    incomeAccounts: SerializedBudgetAccount[];
+    debtCreditDebitGoalAccounts: SerializedBudgetAccount[];
```
[the remark this answers is not in the document]
Comment (mudrila) [reply src/app/components/molecules/BudgetAccount/BudgetAccountList.tsx:10]: Then we wouldn't need `useState` / `useMemo` at all :)

**Review by mudrila — CHANGES_REQUESTED** [review]
Bunch of suggestions / requests for improvements sir, but we're almost there :)

**Comment by Artemh59** [comment]
I hope like this?)

File: src/app/components/pages/Budget/Budget.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import BudgetAccountList from '@/app/components/molecules/BudgetAccount/BudgetAccountList';
+import { auth } from '@/utils/auth';
+import { serializeBudgetAccount } from '@/utils/formatters';
+import prisma from '@/utils/prisma';
+
+export default async function Budget() {
+  const session = await auth();
+  const userId = session?.user?.id;
+
+  const dBaccounts = await prisma.budgetAccount.findMany({
```
Comment (mudrila) [inline src/app/components/pages/Budget/Budget.tsx:12]: Very minor - I'd suggest renaming this just to `budgetAccounts`

File: src/app/components/pages/Budget/Budget.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    where: { userId },
+  });
+  const accounts = dBaccounts.map(serializeBudgetAccount);
```
Comment (mudrila) [inline src/app/components/pages/Budget/Budget.tsx:15]: And then this one to `serializedBudgetAccounts`

File: src/app/components/pages/Budget/Budget.tsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+  const incomeAccounts = accounts.filter(account => account.type === 'Income');
+  const debit = accounts.filter(account => account.type === 'Debit');
+  const credit = accounts.filter(account => account.type === 'Credit');
+  const debt = accounts.filter(account => account.type === 'Debt');
+  const goal = accounts.filter(account => account.type === 'Goal');
+  const spendingCategories = accounts.filter(account => account.type === 'SpendingCategory');
+
+  const grupedAccounts = {
+    incomes: incomeAccounts,
```
Comment (mudrila) [inline src/app/components/pages/Budget/Budget.tsx:25]: You could align object prop names with variables names to leverage prop shorthand usage

```
  const incomes = accounts.filter(account => account.type === 'Income');
  const debits = accounts.filter(account => account.type === 'Debit');
  const credits = accounts.filter(account => account.type === 'Credit');
  const debts = accounts.filter(account => account.type === 'Debt');
  const goals = accounts.filter(account => account.type === 'Goal');
  const spendingCategories = accounts.filter(account => account.type === 'SpendingCategory');

const groupedAccounts = { incomes, debits, credits, goals, spendingCategories }
```

File: src/app/components/pages/Budget/Budget.tsx [hunk 11]
```diff
@@ -0,0 +1,46 @@
+import { Box, Typography } from '@mui/material';
+import BudgetAccountForm from '@/app/components/molecules/BudgetAccount/BudgetAccountForm';
+import BudgetAccountList from '@/app/components/molecules/BudgetAccount/BudgetAccountList';
+import { auth } from '@/utils/auth';
+import { serializeBudgetAccount } from '@/utils/formatters';
+import prisma from '@/utils/prisma';
+
+export default async function Budget() {
+  const session = await auth();
```
Comment (mudrila) [inline src/app/components/pages/Budget/Budget.tsx:9]: I'd personally prefer to see all this logic in `src/app/app/budget/page.tsx` to kinda separate concerns. Data fetching mostly should be happening within `Next.js`-specific components, while displaying and interacting - within UI components. WDYT about such approach?

File: src/app/components/pages/Budget/Budget.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import { Box, Typography } from '@mui/material';
+import BudgetAccountForm from '@/app/components/molecules/BudgetAccount/BudgetAccountForm';
+import BudgetAccountList from '@/app/components/molecules/BudgetAccount/BudgetAccountList';
+import { auth } from '@/utils/auth';
+import { serializeBudgetAccount } from '@/utils/formatters';
+import prisma from '@/utils/prisma';
+
+export default async function Budget() {
```
Comment (mudrila) [inline src/app/components/pages/Budget/Budget.tsx:8]: If we'd follow approach that I'm suggesting - we could remove `async` here and make this component being rendered statically - as opposed to having to `async` components currently

**Review by mudrila — CHANGES_REQUESTED** [review]
Couple minor suggestions, then we'll be good to go sir.

File: src/app/components/pages/Budget/Budget.tsx (hunk 11 above)
Comment (Artemh59) [reply src/app/components/pages/Budget/Budget.tsx:9]: Good approach, I will stick to it in future

**Review by mudrila — APPROVED** [review]
LFG 🚀

**Pull request merged**